pkg/kubelet/kubelet.go

```diff
@@ -1045,7 +1051,6 @@ func (kl *Kubelet) syncPod(pod *api.BoundPod, dockerContainers dockertools.Docke
 			}
 		}
 	}
-	containersToKeep[podInfraContainerID] = empty{}
```

Comment:
I think we're missing removing the POD infra container. We do bellow in the case of a change in the container, but if there is no change we don't seem to be doing it.